mirgecom/__init__.py

```diff
@@ -19,3 +19,30 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+
+import os
+import getpass
+import sys
+from warnings import warn
+
+from mpi4py import MPI
+
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
+
+size = MPI.COMM_WORLD.Get_size()
+
+if size >= 128:
+    for mod in ["pyopencl", "pytools", "loopy"]:
+        if mod in sys.modules:
+            warn("{} already loaded, not adjusting XDG_CACHE_HOME. Please "
+                 "import mirgecom before this module.".format(mod))
+
+    if "XDG_CACHE_HOME" not in os.environ:
+        rank = MPI.COMM_WORLD.Get_rank()
+        username = getpass.getuser()
+
+        xdg_cache_path = "/tmp/{}/xdg-cache-r{}".format(username, rank)
```

Comment:
This is a predictable pattern and hence insecure. Use secure temp file creation.

Reply:
Fixed.